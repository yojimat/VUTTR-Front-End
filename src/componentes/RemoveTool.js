import React from 'react';
import LoadingComponent from "./LoadingComponent";
import Check from "../helpers/icones/Check.svg";
import Close from "../helpers/icones/Close.svg";

const RemoveTool = ({...props }) => {
	const { isLoading,
		nomeTool,
		fecharModal,
		deleteTool,
		respostaFetchUsuario
	} = props;

	return(
		isLoading === true ?
			<LoadingComponent 
				respostaFetchUsuario={ respostaFetchUsuario }
				fecharModal={ () => fecharModal()}
			/>
			:
			<section className="removeTool">
				<p>VocÃª tem certeza que quer remover <strong className="azul">{nomeTool}</strong> da lista?</p>
				<button type="button" 
					onClick={ fecharModal }
					className="br2 ph3 pv2 mb2 dib white mr4 bg-amarelo"
				>
					<span><img src={Close} alt="check" height="20" width="20"/>Cancelar</span>
				</button>
				<button type="submit" 
					onClick={ deleteTool }
					className="br2 ph3 pv2 mb2 dib white bg-vermelho"
				>
					<span><img src={Check} alt="check" height="20" width="20"/>Sim, deletar.</span>
				</button>
			</section>
	);
}
export default RemoveTool;