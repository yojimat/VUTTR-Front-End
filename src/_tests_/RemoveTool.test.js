import React from 'react';
import RemoveTool from '../componentes/RemoveTool';
import renderer from 'react-test-renderer';
import { mount, render } from 'enzyme';

const fecharModal = jest.fn(),
	deleteTool = jest.fn(),
	defaultProps = {
		nomeTool: "",
		fecharModal,
		deleteTool
	};

const RemoveToolCriadorComponents = (props=defaultProps) =>{
	const RemoveToolDefaultComponent = mount(<RemoveTool {...props } />);
	return RemoveToolDefaultComponent;
}

describe(">>>RemoveTool.js", () => {
	it("+RemoveTool renderiza normalmente.", () => {
  		const RemoveToolComponent = renderer.create(<RemoveTool />).toJSON();

  		expect(RemoveToolComponent).toMatchSnapshot();
	});

	it("+Verificando default props", () => {
		const RemoveToolComponent = RemoveToolCriadorComponents();

		expect(RemoveToolComponent.prop('nomeTool')).toEqual("");
		expect(RemoveToolComponent.prop('fecharModal')).toEqual(fecharModal);
		expect(RemoveToolComponent.prop('deleteTool')).toEqual(deleteTool);
		expect.assertions(3);
	});

	it("+Verificando valor nomeTool", ()=>{
		const props = {
			nomeTool: "GitHub"
		},
		RemoveToolComponentPTag = RemoveToolCriadorComponents(props).find("p");

		expect(RemoveToolComponentPTag).toHaveLength(1);
		expect(RemoveToolComponentPTag.text()).toMatch(props.nomeTool);
		expect.assertions(2);
	});

	it("+Verificando evento fechar Modal", () =>{
		const RemoveToolComponentCancelar = RemoveToolCriadorComponents().find('p + button');
    	RemoveToolComponentCancelar.simulate('click', fecharModal);
    	expect(fecharModal).toHaveBeenCalled();
	});

	it("+Verificando evento deleteTool", () =>{
		const RemoveToolComponentDel = RemoveToolCriadorComponents().find('button + button');
    	RemoveToolComponentDel.simulate('click', deleteTool);
    	expect(deleteTool).toHaveBeenCalled();
	});

	it("+Verificando evento isLoading", () => {
		const propsLoading = {
			isLoading: true,
			respostaFetchUsuario: "",
			fecharModal
		};
		const removeToolLoading = render(<RemoveTool{...propsLoading } />);
		
		expect(removeToolLoading).toMatchSnapshot();
	});
});