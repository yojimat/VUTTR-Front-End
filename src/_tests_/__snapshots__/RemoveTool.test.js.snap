// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`>>>RemoveTool.js +RemoveTool renderiza normalmente. 1`] = `
<section
  className="removeTool"
>
  <p>
    VocÃª tem certeza que quer remover 
    <strong
      className="azul"
    />
     da lista?
  </p>
  <button
    className="br2 ph3 pv2 mb2 dib white mr4 bg-amarelo"
    type="button"
  >
    <span>
      <img
        alt="check"
        height="20"
        src="Close.svg"
        width="20"
      />
      Cancelar
    </span>
  </button>
  <button
    className="br2 ph3 pv2 mb2 dib white bg-vermelho"
    type="submit"
  >
    <span>
      <img
        alt="check"
        height="20"
        src="Check.svg"
        width="20"
      />
      Sim, deletar.
    </span>
  </button>
</section>
`;

exports[`>>>RemoveTool.js +Verificando evento isLoading 1`] = `
initialize {
  "0": Object {
    "attribs": Object {
      "class": "loader",
    },
    "children": Array [],
    "name": "div",
    "namespace": "http://www.w3.org/1999/xhtml",
    "next": null,
    "parent": null,
    "prev": null,
    "root": Object {
      "attribs": Object {},
      "children": Array [
        [Circular],
      ],
      "name": "root",
      "namespace": "http://www.w3.org/1999/xhtml",
      "next": null,
      "parent": null,
      "prev": null,
      "type": "root",
      "x-attribsNamespace": Object {},
      "x-attribsPrefix": Object {},
    },
    "type": "tag",
    "x-attribsNamespace": Object {
      "class": undefined,
    },
    "x-attribsPrefix": Object {
      "class": undefined,
    },
  },
  "_root": [Circular],
  "length": 1,
  "options": Object {
    "decodeEntities": true,
    "normalizeWhitespace": false,
    "withDomLvl1": true,
    "xml": false,
  },
}
`;
